### PR TITLE
common/admin_socket: Add doxygen for call and call_async

### PR DESCRIPTION
**Pull request by aclamk**
This is only doxygen, an attempt to solidify interface contract.

Signed-off-by: Adam Kupczyk <akupczyk@redhat.com>

### CODE REVIEW

File: src/common/admin_socket.h [hunk 1]
```diff
@@ -36,15 +36,57 @@ inline constexpr auto CEPH_ADMIN_SOCK_VERSION = "2"sv;
...
+  /**
+   * Handler for admin socket commands, synchronous version
+   *
+   * Executes action associated with admin command and returns byte-stream output \ref out.
+   * There is no restriction on output. Each handler defines output semantics.
+   * Typically output is textual representation of some ceph's internals.
+   * Handler should use provided formatter \ref f if structuralized output is being produced.
+   *
+   * Params:
+   *   command - string matching constant part of cmddesc in \ref AdminSocket::register_command
```
Comment (tchaikov) [inline src/common/admin_socket.h]: could you use 
```
@param command string matching constant part of cmddesc in @ref AdminSocket::register_command
...
@retval 0 success, errss is ignored and does not contribute to output
@retval <0 error code, errss is prepended
```

where "param" and "ref" commands are used.

File: src/common/admin_socket.h [hunk 2]
```diff
@@ -36,15 +36,56 @@ inline constexpr auto CEPH_ADMIN_SOCK_VERSION = "2"sv;
...
+  /**
+   * @brief
+   * Handler for admin socket commands, synchronous version
+   *
+   * Executes action associated with admin command and returns byte-stream output @ref out.
+   * There is no restriction on output. Each handler defines output semantics.
+   * Typically output is textual representation of some ceph's internals.
+   * Handler should use provided formatter @ref f if structuralized output is being produced.
+   *
+   * @param command[in] String matching constant part of cmddesc in @ref AdminSocket::register_command
```
Comment (tchaikov) [inline src/common/admin_socket.h]: s/ref/c/

see http://www.doxygen.nl/manual/commands.html#cmdc and http://www.doxygen.nl/manual/commands.html#cmdref .

File: src/common/admin_socket.h [hunk 3]
```diff
@@ -36,15 +36,56 @@ inline constexpr auto CEPH_ADMIN_SOCK_VERSION = "2"sv;
...
+   * @param cmdmap[in]  Parameters extracted from argument part of cmddesc in @ref AdminSocket::register_command
+   * @param f[in]       Formatter created according to requestor preference, used by `ceph --format`
+   * @param errss[out]  Error stream, containing details in case of execution failure
+   * @param out[out]    Produced output
+   *
+   * @return 0 Success, errss is ignored and does not contribute to output
```
Comment (tchaikov) [inline src/common/admin_socket.h]: s/return/retval/ see http://www.doxygen.nl/manual/commands.html#cmdretval

File: src/common/admin_socket.h (hunk 2 above)
Comment (aclamk) [reply src/common/admin_socket.h]: @tchaikov Done.I still left \ref in places where it referred methods. You like it?

**Pull request merged**